utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,523 @@
+		Error::Crypto(e)
+	}
+}
+
+impl From<EPM::unsigned::MinerError> for Error {
+	fn from(e: EPM::unsigned::MinerError) -> Error {
+		Error::PalletMiner(e)
+	}
+}
+
+impl From<EPM::ElectionError> for Error {
+	fn from(e: EPM::ElectionError) -> Error {
+		Error::PalletElection(e)
+	}
+}
+
+impl From<EPM::FeasibilityError> for Error {
+	fn from(e: EPM::FeasibilityError) -> Error {
+		Error::PalletFeasibility(e)
+	}
+}
+
+impl std::fmt::Display for Error {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		<Error as std::fmt::Debug>::fmt(self, f)
+	}
+}
+
+#[derive(Debug, Clone, StructOpt)]
+enum Command {
+	/// Monitor for the phase being signed, then compute.
+	Monitor(MonitorConfig),
+	/// Just compute a solution now, and don't submit it.
+	DryRun(DryRunConfig),
+	/// Provide a solution that can be submitted to the chian as an emergency response.
+	EmergencySolution,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct MonitorConfig {
+	/// They type of event to listen to.
+	///
+	/// Typically, finalized is safer and there is no chance of anything going wrong, but it can be
+	/// slower. It is recommended if the duration of the signed phase is longer than the a
+	#[structopt(long, default_value = "head", possible_values = &["head", "finalized"])]
+	listen: String,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct DryRunConfig {
+	/// The block hash at which scraping happens. If none is provided, the latest head is used.
+	#[structopt(long)]
+	at: Option<Hash>,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct SharedConfig {
+	/// The ws node to connect to.
+	#[structopt(long, default_value = DEFAULT_URI)]
+	uri: String,
+
+	/// The file from which we read the account seed.
+	#[structopt(long)]
+	account_seed: PathBuf,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct Opt {
+	/// The ws node to connect to.
+	#[structopt(flatten)]
+	shared: SharedConfig,
+
+	#[structopt(subcommand)]
+	command: Command,
+}
+
+/// Some information about the signer. Redundant at this point, but makes life easier.
+#[derive(Clone)]
+struct Signer {
+	/// The account id.
+	account: AccountId,
+	/// The full crypto key-pair.
+	pair: Pair,
+	/// The raw uri read from file.
+	uri: String,
+}
+
+/// Build the `Ext` at `hash` with all the data of `ElectionProviderMultiPhase` and `Staking`
+/// stored.
+async fn create_election_ext<T: EPM::Config, B: BlockT>(
+	uri: String,
+	at: Option<B::Hash>,
+	with_staking: bool,
+) -> Result<Ext, Error> {
+	use frame_support::{storage::generator::StorageMap, traits::PalletInfo};
+	let system_block_hash_key = <frame_system::BlockHash<T>>::prefix_hash();
+
+	Builder::<B>::new()
+		.mode(Mode::Online(OnlineConfig {
+			transport: uri.into(),
+			at,
+			modules: if with_staking {
+				vec![
+					<T as frame_system::Config>::PalletInfo::name::<EPM::Pallet<T>>()
+						.expect("Pallet always has name; qed.")
+						.to_string(),
+					<T as frame_system::Config>::PalletInfo::name::<pallet_staking::Pallet<T>>()
```

Comment:
If we are grabbing the snapshot I think we need to make the max rpc payload param on the node is raised

Reply:
also would be guranteed failure on public rpc nodes (like the default uri) assumng the have not raise the max rpc payload param

Reply:
a note about the max rpc payload could be useful in this cli help or maybe higher level docs

Reply:
yeah for sure, this tool needs that flag.

Reply:
> also would be guranteed failure on public rpc nodes (like the default uri) assumng the have not raise the max rpc payload param

unfortunate, but true.